### ISSUE TITLE
printing of objects should compose better

### ISSUE THREAD

**Issue by tthsqe12** [oscar-system/Oscar.jl#347]
The current printing of, say, ideals is fine in isolation:
```
julia> R,(x,y,z)=QQ["x","y","z"]
(Multivariate Polynomial Ring in x, y, z over Rational Field, fmpq_mpoly[x, y, z])

julia> ideal(R,[x,y])
ideal generated by: x, y
```

But, when it is used in the context of other printing, it becomes a mess:
```
julia> (ideal(R,[x,y]), x, y)
(ideal generated by: x, y, x, y)

julia> (ideal(R,[x,y,x]),y)
(ideal generated by: x, y, x, y)
```

**Comment by fingolfin** [comment]
How about `ideal generated by [x, y]` ?

**Comment by tthsqe12** [comment]
Since one generally inputs these things as `ideal(R, [x, y])`, I am inclined to have it print as `ideal([x, y])` or `ideal(x, y)`. Any objections? @wdecker

**Comment by wdecker** [comment]
I would prefer less brackets: ideal(x,y)

**Comment by thofma** [comment]
While you are at it, please improve the printing of `Qab*`.

**Comment by simonbrandhorst** [comment]
#2148

**Comment by simonbrandhorst** [comment]
closed via #2148
